examples/python-loganalysis/loganalysis.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import re
+import requests
+import json
+
+prelines = 10
+postlines = 10
+
+def find_errors_in_log_file():
+  if len(sys.argv) < 2:
+    print("Usage: python loganalysis.py <filename>")
+    return
+
+  log_file_path = sys.argv[1]
+  with open(log_file_path, 'r') as log_file:
+    log_lines = log_file.readlines()
+
+  error_lines = []
+  for i, line in enumerate(log_lines):
+    if re.search('error', line, re.IGNORECASE):
+      error_lines.append(i)
+
+  error_logs = []
+  for error_line in error_lines:
+    start_index = max(0, error_line - prelines)
+    end_index = min(len(log_lines), error_line + postlines)
+    error_logs.extend(log_lines[start_index:end_index])
```

Comment:
Just a suggestion, I think you could do this all in one loop

```suggestion
error_logs = []
    for i, line in enumerate(log_lines):
        if "error" in line.lower():
            start_index = max(0, i - prelines)
            end_index = min(len(log_lines), i + postlines + 1)
            error_logs.extend(log_lines[start_index:end_index])
```